### PR TITLE
Add support for dropping column in MongoDB

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java [hunk 1]
```diff
@@ -210,6 +210,26 @@ public void addColumn(SchemaTableName schemaTableName, ColumnMetadata columnMeta
         tableCache.invalidate(schemaTableName);
     }
 
+    public void dropColumn(SchemaTableName schemaTableName, String columnName)
+    {
+        String schemaName = toRemoteSchemaName(schemaTableName.getSchemaName());
```
Comment (kokosing) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java:215]: s/schemaName/remoteSchemaName and same for tableName below?

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java [hunk 2]
```diff
@@ -210,6 +210,26 @@ public void addColumn(SchemaTableName schemaTableName, ColumnMetadata columnMeta
...
+        String tableName = toRemoteTableName(schemaName, schemaTableName.getTableName());
+
+        Document metadata = getTableMetadata(schemaName, tableName);
+
+        List<Document> columns = getColumnMetadata(metadata).stream()
+                .filter(doc -> !doc.getString(FIELDS_NAME_KEY).equals(columnName))
```
Comment (kokosing) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java:221]: s/doc/document

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java [hunk 3]
```diff
@@ -210,6 +210,26 @@ public void addColumn(SchemaTableName schemaTableName, ColumnMetadata columnMeta
...
+                .collect(toImmutableList());
+
+        metadata.append(FIELDS_KEY, columns);
+
+        MongoDatabase db = client.getDatabase(schemaName);
```
Comment (kokosing) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java:226]: s/db/database

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java [hunk 4]
```diff
@@ -210,6 +210,26 @@ public void addColumn(SchemaTableName schemaTableName, ColumnMetadata columnMeta
...
+        String schemaName = toRemoteSchemaName(schemaTableName.getSchemaName());
+        String tableName = toRemoteTableName(schemaName, schemaTableName.getTableName());
+
+        Document metadata = getTableMetadata(schemaName, tableName);
+
+        List<Document> columns = getColumnMetadata(metadata).stream()
+                .filter(doc -> !doc.getString(FIELDS_NAME_KEY).equals(columnName))
+                .collect(toImmutableList());
+
+        metadata.append(FIELDS_KEY, columns);
```
Comment (kokosing) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java:224]: does "append" replace the actual columns?

**Review by kokosing — APPROVED** [review]

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java (hunk 4 above)
Comment (ebyhr) [reply plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java:224]: Yes, the internal implementation is `LinkedHashMap.put()` method.

**Pull request merged**